platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/SecurityUtils.java

```diff
@@ -0,0 +1,217 @@
+        }
+
+        KeyStore keyStore = getSystemKeyStore();
+        String alias = null;
+        Certificate cert;
+        try {
+            if (keyStore.size() == 1) {
+                alias = keyStore.aliases().nextElement();
+            } else if (keyStore.size() > 1) {
+                alias = getCertificatAlias();
+            }
+            cert = keyStore.getCertificate(alias);
+        } catch (KeyStoreException e) {
+            LOGGER.error("Unable to get certificate for alias [{}]", alias, e);
+            return null;
+        }
+
+        if (cert == null) {
+            LOGGER.error("Unable to get certificate for alias [{}]", alias);
+            return null;
+        }
+
+        PKIAuthenticationTokenFactory pkiTokenFactory = createPKITokenFactory();
+        PKIAuthenticationToken pkiToken = pkiTokenFactory
+                .getTokenFromCerts(new X509Certificate[] {(X509Certificate) cert},
+                        PKIAuthenticationToken.DEFAULT_REALM);
+        if (pkiToken != null) {
+            SecurityManager securityManager = getSecurityManager();
+            if (securityManager != null) {
+                synchronized (lock) {
+                    try {
+                        cachedSystemSubject = securityManager.getSubject(pkiToken);
+                    } catch (SecurityServiceException sse) {
+                        LOGGER.error("Unable to request subject for system user.", sse);
+                    }
+                }
+            }
+        }
+        return cachedSystemSubject;
+    }
+
+    public SecurityManager getSecurityManager() {
+        BundleContext context = getBundleContext();
+        if (context != null) {
+            ServiceReference securityManagerRef = context
+                    .getServiceReference(SecurityManager.class);
+            return (SecurityManager) context.getService(securityManagerRef);
+        }
+        LOGGER.warn("Unable to get Security Manager");
+        return null;
+    }
+
+    public BundleContext getBundleContext() {
+        Bundle bundle = FrameworkUtil.getBundle(SecurityUtils.class);
+        if (bundle != null) {
+            return bundle.getBundleContext();
+        }
+        return null;
+    }
+
+    public PKIAuthenticationTokenFactory createPKITokenFactory() {
+        PKIAuthenticationTokenFactory pkiTokenFactory = new PKIAuthenticationTokenFactory();
+        pkiTokenFactory.init();
+        return pkiTokenFactory;
+    }
+
+    private String getCertificatAlias() {
+        BundleContext context = getBundleContext();
+        if (context != null) {
+            ServiceReference configAdminRef = context.getServiceReference(ConfigurationAdmin.class);
+            ConfigurationAdmin configAdmin = (ConfigurationAdmin) context
+                    .getService(configAdminRef);
+            Configuration config;
+            try {
+                config = configAdmin.getConfiguration("ddf.platform.config", null);
```

Comment:
It's not guaranteed to be the right name but it's the best we have right now. This should be updated next sprint when we will have a place that is guaranteed to have the right value. 
Since we are passing in null as the location, it should not change its bundle binding.